lib/Client.php

```diff
@@ -169,8 +175,17 @@ public function makeRequest($method, $url, $body = null, $headers = null)
         $responseHeaders = array_map('trim', $responseHeaders);
 
         curl_close($curl);
+     
+        $response = new Response($statusCode, $responseBody, $responseHeaders);
+
+        if ($statusCode == 429 && $retryOnLimit) {
+            $headers = $response->headers(true);
+            $sleepDurations = $headers['X-Ratelimit-Reset'] - time();
```

Comment:
Please explain the idea behind this algorithm.

I want to make sure we don't overload our servers when something has gone wrong. I was hoping for a backoff strategy that eventually fails after a certain period of time.

Reply:
Every request we made to sendgrid will have X-Ratelimit-Reset in the header, where it tell us when the rate limit is reset in sendgrid.
So when ever we have error 429(rate limit reached) we use X-Ratelimit-Reset to determine how long the script need to wait before make another call to sendgrid.
This way we don't use a random time or predefined time.